support/globalconfig/network.go

```diff
@@ -25,12 +28,22 @@ func ReconcileNetworkConfig(cfg *configv1.Network, hcp *hyperv1.HostedControlPla
 	for _, entry := range hcp.Spec.Networking.ClusterNetwork {
 		hostPrefix := uint32(entry.HostPrefix)
 		if hostPrefix == 0 {
-			hostPrefix = defaultHostPrefix
+			ipv4, err := util.IsIPv4(entry.CIDR.String())
+			if err != nil {
+				fmt.Printf("error checking the ClusterNetwork CIDR %s: %v\n", entry.CIDR.String(), err)
+				continue
+			}
+
+			if ipv4 {
+				hostPrefix = defaultIPv4HostPrefix
+			} else {
+				hostPrefix = defaultIPv6HostPrefix
+			}
+			clusterNetwork = append(clusterNetwork, configv1.ClusterNetworkEntry{
+				CIDR:       entry.CIDR.String(),
+				HostPrefix: hostPrefix,
+			})
```

Comment:
why was this moved inside the `if hostPrefix == 0 {}`

Reply:
Oh, good catch, you're right, let me fix it.